atlasdb-cassandra/src/main/java/com/palantir/atlasdb/keyvalue/cassandra/pool/CassandraService.java

```diff
@@ -68,14 +78,30 @@
 
     private List<InetSocketAddress> cassandraHosts;
 
+    private volatile Set<InetSocketAddress> localHosts;
+    private final Supplier<Optional<HostLocation>> myLocationSupplier;
+
+    private final Counter randomHostsSelected;
+    private final Counter localHostsSelected;
+
+    private final Random random;
+
     public CassandraService(MetricsManager metricsManager, CassandraKeyValueServiceConfig config, Blacklist blacklist) {
         this.metricsManager = metricsManager;
+        this.randomHostsSelected = metricsManager.getTaggedRegistry().counter(MetricName.builder()
+                .safeName(MetricRegistry.name(CassandraService.class, "randomHostsSelected")).build());
+        this.localHostsSelected = metricsManager.getTaggedRegistry().counter(MetricName.builder()
+                .safeName(MetricRegistry.name(CassandraService.class, "localHostsSelected")).build());
         this.config = config;
         this.blacklist = blacklist;
+        this.localHosts = ImmutableSet.of();
+        this.myLocationSupplier = new HostLocationSupplier(this::getSnitch, config.overrideHostLocation());
+        this.random = new Random();
```

Comment:
as before - you can initialize this inline - doesn't have to happen in the constructor. Same with the localHosts bit.

Reply:
in general, prefer to initialize things in the constructor _that need to be_ initialized there